### DISCUSSION TITLE
[2024-03-18] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#9460] [Weekly Demo Day]
@farmaazon 

* #9399 Clicking on Text widget opens drop-down (if any), and when the user starts typing anything, the filter is applied. Enter/clicking app background will accept the input as is, and clicking on drop down entry inserts it and finishes editing. `Esc` restore the previous state.
* #9366 with @kazcw and @Frizi: Fixed a nasty bug with AST adding space in places it should not (causing problems, when the space happens to be between `on=` and arg value).

[Screencast from 2024-03-13 15-20-17.webm](https://github.com/enso-org/enso/assets/3919101/112ce65a-a8c6-4818-b8b8-9f493caf9c81)

@Frizi 
* #9336 - regression caused by previous iteration's work
* Working on #8520: the partial work is already merged (the necessary refactoring of tests): #9403

@vitvakatu 
* #9365
* #9401 - avoiding another set of transient issues
* #9302 - :tada: 
* #9335 - a small visual improvement, fixing #9333
* #9290

https://github.com/enso-org/enso/assets/6566674/43984978-004b-4ed2-a55e-de9302dc1fdd

https://github.com/enso-org/enso/assets/6566674/46a6de40-cbd0-427e-942d-b58ad4a9fd33

https://github.com/enso-org/enso/assets/6566674/705e23c2-0ee0-40b0-9919-9c320f8788c5

**Comment by kazcw** [comment]
# Major UI features (3+ points)

### [Scrollbars](https://github.com/enso-org/enso/issues/8617)

https://github.com/enso-org/enso/assets/1047859/d78cb46c-478d-4520-b4db-62b103ac791e

### [Conditional ports](https://github.com/enso-org/enso/issues/9212)

https://github.com/enso-org/enso/assets/1047859/68ac1953-c8b1-4e51-8c4c-211595e0c034

### [Selection layer](https://github.com/enso-org/enso/issues/8938)

https://github.com/enso-org/enso/assets/1047859/190006df-1402-4a5f-9117-f43788fa4187

### [Pan to CB](https://github.com/enso-org/enso/issues/8618)

https://github.com/enso-org/enso/assets/1047859/a9f18df5-c0ca-426c-959a-bda5cd077541

# Minor UI features (<3 points)

<table>
<tr><td>

[JSON viz](https://github.com/enso-org/enso/issues/9226)

https://github.com/enso-org/enso/assets/1047859/1adc1acb-ebd2-42b7-915d-98c486daf895

</td><td>

[Hover menu](https://github.com/enso-org/enso/issues/9211)

https://github.com/enso-org/enso/assets/1047859/364b72fb-8198-4fb2-a490-17fab0286766

</td></tr>
<tr><td>

[Remove-warnings](https://github.com/enso-org/enso/issues/9226)

https://github.com/enso-org/enso/assets/1047859/1b00a62e-8f3b-4697-af10-34a4b93ca8ff

</td><td>

[Zoom steps](https://github.com/enso-org/enso/issues/9177)

https://github.com/enso-org/enso/assets/1047859/f6abb330-2fed-4f1f-aee3-951ac7dc2799

</td></tr>
</table>

# Other
- <img width="16" src="https://github.com/enso-org/enso/assets/1047859/218ee58c-e461-4ebd-a6e4-e0b75ce33b0b" /> #9372 <img width="16" src="https://github.com/enso-org/enso/assets/1047859/218ee58c-e461-4ebd-a6e4-e0b75ce33b0b" />
- [Editing node maintains record-mode state](https://github.com/enso-org/enso/issues/7976)
- #9261
- #9379
- #9328 support for upcoming features...

**Comment by mwu-tow** [comment]
Minor improvements:
* Allow requesting a clean build through the web dispatch https://github.com/enso-org/enso/pull/9270
* Keep directory for the uploaded playwright reports https://github.com/enso-org/enso/pull/9323

Engine-specific:
* https://github.com/enso-org/enso/pull/9295
  * Split the Engine CI job into a few smaller ones.
  * Gives significant start-to-end time reduction.
* [in progress] Removing discrepencies between macOS and non-macOS-intended CI checks https://github.com/enso-org/enso/pull/9309

Old GUI removal followup — removing more of the old code and related updates to the build script.
* https://github.com/enso-org/enso/pull/9342
* https://github.com/enso-org/enso/pull/9407
* https://github.com/enso-org/enso/pull/9376

Ongoing monitoring of the CI failures and reporting issues.
* Still new random test regressions keep popping up.
* Runner's load issues.

**Comment by hubertp** [comment]
Relevant backend work:

- Pending: Instrumentation, visualization and (autoscoped) constructors https://github.com/enso-org/enso/pull/9452
- Pending: Speedup visualization and widget responses https://github.com/enso-org/enso/pull/9371
- Local project manager work https://github.com/enso-org/enso/pull/9245
- Pending: Table.join - aka access arguments when widgets annotations are computed
- Discussed: y.js server in the JVM
- Pending: Fixing reexported symbols: https://github.com/enso-org/enso/pull/9454
- Pending: Make FQN and imports consistent https://github.com/enso-org/enso/pull/9367
